design/one-pager-terrajet-metadata-extraction.md

```diff
@@ -0,0 +1,272 @@
+            admin_users: '- (Optional) List of email addresses of admin users.'
+            backup_blob_container_uri: '- (Optional) URI and SAS token for a blob container to store backups.'
+            enable_power_bi_service: '- (Optional) Indicates if the Power BI service is allowed to access or not.'
+            # exported attributes appear under the "exportedAttributes" map (as a block)
+            exportedAttributes:
+                id: '- The ID of the Analysis Services Server.'
+                server_full_name: '- The full name of the Analysis Services Server.'            
+            # parameters with block values are maps
+            ipv4_firewall_rule:
+                name: '- (Required) Specifies the name of the firewall rule.'
+                # if the block-valued parameter has itself a description, it appears under "nodeText"
+                # We assume "nodeText" is not a valid parameter/attribute name
+                nodeText: '- (Optional) One or more ipv4_firewall_rule block(s) as defined below.'
+                range_end: '- (Required) End of the firewall rule range as IPv4 address.'
+                range_start: '- (Required) Start of the firewall rule range as IPv4 address.'
+            location: '- (Required) The Azure location where the Analysis Services Server exists. Changing this forces a new resource to be created.'
+            name: '- (Required) The name of the Analysis Services Server. Changing this forces a new resource to be created.'
+            querypool_connection_mode: '- (Optional) Controls how the read-write server is used in the query pool. If this value is set to All then read-write servers are also used for queries. Otherwise with ReadOnly these servers do not participate in query operations.'
+            resource_group_name: '- (Required) The name of the Resource Group in which the Analysis Services Server should be exist. Changing this forces a new resource to be created.'
+            sku: '- (Required) SKU for the Analysis Services Server. Possible values are: D1, B1, B2, S0, S1, S2, S4, S8, S9, S8v2 and S9v2.'
+            timeouts:
+                create: '- (Defaults to 30 minutes) Used when creating the Analysis Services Server.'
+                delete: '- (Defaults to 30 minutes) Used when deleting the Analysis Services Server.'
+                read: '- (Defaults to 5 minutes) Used when retrieving the Analysis Services Server.'
+                update: '- (Defaults to 30 minutes) Used when updating the Analysis Services Server.'
+        # import statement scraped from the Terraform registry, if available
+        # Can be used for advanced purposes, such as constructing resource config "ExternalName.GetIDFn" functions, etc.
+        importStatements:
+            - terraform import azurerm_analysis_services_server.server /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourcegroup1/providers/Microsoft.AnalysisServices/servers/server1
+```
+
```

Comment:
Could you elaborate on why the chosen method is better than these alternatives? This would help us reason about them more deliberately when we want to make changes to the pipeline.

Reply:
Added clarifications on the choices we made here.